### ISSUE TITLE
OrientDB 2.2.3 - noTx.drop() leads to NPE in OrientBaseGraph

### ISSUE THREAD

**Issue by Jotschi** [orientechnologies/orientdb#6336]
## Expected behavior and actual behavior

I'm currently trying to upgrade from OrientDB 2.1.x to 2.2.x unfortunately a strange NPE is causing a lot of test errors.

```
19:53:40.980 [Cherished Rayquaza] ERROR [vert.x-worker-thread-5] [AbstractDatabase.java:129] - Error while handling no-transaction.
java.lang.NullPointerException: null
	at com.tinkerpop.blueprints.impls.orient.OrientBaseGraph.getVertices(OrientBaseGraph.java:893) ~[orientdb-graphdb-2.2.3.jar:2.2.3]
	at com.gentics.mesh.graphdb.OrientDBDatabase.getVertices(OrientDBDatabase.java:199) ~[classes/:na]
	at com.gentics.mesh.core.data.root.impl.UserRootImpl.findMeshAuthUserByUuid(UserRootImpl.java:97) ~[classes/:na]
	at com.gentics.mesh.auth.MeshJWTAuthProvider.lambda$1(MeshJWTAuthProvider.java:66) ~[classes/:na]
	at com.gentics.mesh.graphdb.spi.AbstractDatabase.lambda$1(AbstractDatabase.java:121) ~[classes/:na]
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$17(ContextImpl.java:296) ~[vertx-core-3.2.1.jar:na]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[na:1.8.0_74]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[na:1.8.0_74]
```

In OrientBaseGraph:892 clazz is `null`.
```
    final OClass clazz = getDatabase().getMetadata().getImmutableSchemaSnapshot().getClass(label);
```

## Steps to reproduce the problem

Unfortunately I was unable to create a reproducer. I tried to isolate the code but as soon as I removed it from the vert.x thread context the issue could not be reproduced.

### Our Setup

In our test suite we are executing a lot of unit tests which access the OrientDBGraph. Some transactions we use are created within a thread pool that is most likely not recreated in between tests. The transactions are fetched from a central `OrientGraphFactory` and closed via the tx.shutdown() call. Instead of closing the factory we reuse the previously created OrientGraphFactory for each test case. In between tests the Database is purged via:

```
   OrientGraphNoTx noTx = factory.getNoTx();
   try {
	noTx.drop();
   } finally {
	noTx.shutdown();
  }
```

For 2.1.x this was sufficient but in 2.2.x the NPE is occurring. 

### Clue 1:

If I change our test setup slightly and *close* the factory instead of reusing it for the tests. In that case the tests pass.

### Clue 2: 

The tests also pass if I replace the noTx.drop() call with:

```
try (Trx tx = trx()) {
	tx.getGraph().e().removeAll();
	tx.getGraph().v().removeAll();
	tx.success();
}
```

I believe that the issue is located around `noTx.drop()`.

### Runninng Mode
- [x] Embedded, using MEMORY access mode

### OrientDB Version
- [x] v2.2.3

### Operating System
- [x] Linux

### Java Version
- [x] 8

**Comment by luigidellaquila** [comment]
Hi @Jotschi 

From your stack trace it seems that you are manually invoking graph.getVertices(), probably when some classes were already deleted.
Could this be a MultiThread issue in your test?

Anyway, I'll fix the NPE

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Ok, I pushed a fix on both 2.2.x and develop branches, now the ugly NPE should not happen anymore (but for sure you will have a different - explicit - error if a class does not exist)

Thanks

Luigi

**Comment by Jotschi** [comment]
@luigidellaquila Thanks. Could you perhaps also check whether OrientGraphNoTx#drop() is clearing all caches, stores etc? Luca fixed a NPE in drop() a while ago for 2.2.x see #6171 - Maybe thats related?
Otherwise the issue can be closed. I already use the workaround I described.
